analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,248 @@
+                    {
+                        if (IsInvalidKey(invocation.ArgumentList.Arguments[0], programState))
+                        {
+                            context.AddLocation(invocation.GetLocation());
+                        }
+                        else if (IsInvalidIV(invocation.ArgumentList.Arguments[1], programState))
+                        {
+                            context.AddLocation(invocation.GetLocation());
+                        }
+                    }
+                }
+
+                return programState;
+            }
+
+            private ProgramState ArrayCreationPostProcess(ArrayCreationExpressionSyntax arrayCreation, ProgramState programState) =>
+                semanticModel.GetTypeInfo(arrayCreation).Type.Is(KnownType.System_Byte_Array) && programState.HasValue
+                    ? programState.SetConstraint(programState.PeekValue(), ConstantByteArraySymbolicValueConstraint.Constant)
+                    : programState;
+
+            private bool IsInvalidKey(ArgumentSyntax argument, ProgramState programState) =>
+                argument.Expression is MemberAccessExpressionSyntax memberAccess
+                && memberAccess.NameIs("Key")
+                && programState.GetSymbolValue(semanticModel.GetSymbolInfo(memberAccess.Expression).Symbol) is {} symbolicValue
+                && HasNotInitializedKeyConstraint(symbolicValue, programState);
+
+            private bool IsInvalidIV(ArgumentSyntax argument, ProgramState programState) =>
+                argument.Expression switch
+                {
+                    MemberAccessExpressionSyntax memberAccess => memberAccess.NameIs("IV")
+                                                                 && programState.GetSymbolValue(semanticModel.GetSymbolInfo(memberAccess.Expression).Symbol) is {} symbolicValue
+                                                                 && HasNotInitializedIVConstraint(symbolicValue, programState),
+                    IdentifierNameSyntax identifier => programState.GetSymbolValue(semanticModel.GetSymbolInfo(identifier).Symbol) is {} symbolicValue
+                                                       && programState.HasConstraint(symbolicValue, ConstantByteArraySymbolicValueConstraint.Constant),
+                    _ => false
+                };
+
+            private SymbolicValue GetSymbolicValue(InvocationExpressionSyntax invocation, ProgramState programState)
+            {
+                var invocationTarget = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
+                var targetSymbol = semanticModel.GetSymbolInfo(invocationTarget).Symbol;
+
+                return programState.GetSymbolValue(targetSymbol);
+            }
+
+            private bool IsIVMemberAccess(MemberAccessExpressionSyntax memberAccess) =>
+                memberAccess.NameIs("IV")
+                && semanticModel.GetSymbolInfo(memberAccess).Symbol.ContainingType.IsAny(VulnerableTypes);
```

Comment:
We need a null check here, as unresolved types can cause symbol to be null:
```C#
using (FakeUnresolvedType sa = SymmetricAlgorithm.Create("AES"))    // Error [CS0246]
{
    sa.IV = initializationVectorConstant;
}
```

Reply:
Coverage seems happy, but do we have another test case for this?

Reply:
I don't understand why coverage is happy since we don't cover this case. I even attached the debugger to double check 😄 

I've added the suggested test case.